### ISSUE TITLE
Steam Login Support?

### ISSUE THREAD

**Issue by bloggingninja** [Healix/Gw2Launcher#187]
Is there any possibility of adding support for the steam login to gw2Launcher?

you can get the launcher with "login using steam" by using the "-provider Steam" Argument when launching gw2 but it says there's a problem with steam even if I have steam running (no issue if launched from within steam). 

I don't know if this request is possible but I figured I'd ask.

**Comment by Healix** [comment]
That option basically just swaps the login interface - it doesn't work because it wasn't launched through Steam and thus isn't properly connected to it. Is it possible to support multiple Steam accounts? No. A single Steam account? Yes, but worst case it'd be like launching GW2 outside of Gw2Launcher and only having Gw2Launcher track it.

**Comment by Healix** [comment]
Turns out this is trivial. I was assuming Steam was injecting, but that's only the initial overlay. Here's how to do it right now:

Open notepad, paste the following:

```
set SteamAppId=1284210
start Gw2Launcher.exe
```

Save it as Gw2Launcher.bat in the same folder as Gw2Launcher.exe. Run the batch file to start Gw2Launcher. You can now use "-provider Steam" to login through Steam.

**Comment by Christoferthz** [comment]
So, can we run Gw 2 Steam (with old non-steam account, using -provider Portal ) and gw 2 launcher and other non-steam accounts at the same time?

**Comment by Healix** [comment]
> So, can we run Gw 2 Steam (with old non-steam account, using -provider Portal ) and at the same time in gw 2 launcher other non-steam accounts?

Yes. You'd just need to use the -shareArchive option (in Steam) to allow multiple clients to share the same files.

**Comment by Christoferthz** [comment]
Well it worked for me, is there any way to have only one gw 2 installed? Right now I have two, I'm afraid of deleting something and the 50 accounts I have will stop working, and another question, could gw 2 steam run (without a steam account) when I press my account from the gw 2 launcher list?

**Comment by writzx** [comment]
Will there be a way to start the non-steam account via Gw2Launcher and yet display the game on steam profile status?

Similar to what we can do when we run the game through steam with `-provider Portal` option but through Gw2Launcher.

**Comment by Healix** [comment]
> is there any way to have only one gw 2 installed?

The Steam and regular version of GW2 are identical. You can either make Steam use your original GW2 folder or your original GW2 use Steam. The simplest option would be deleting your original folder, then Gw2Launcher will ask where GW2 is on the next launch, so you select the Steam folder.

> could gw 2 steam run (without a steam account) when I press my account from the gw 2 launcher list?

Through Steam (Gw2Launcher tells Steam to launch GW2)? That would be like running GW2 outside of Gw2Launcher - any of the configurations from Gw2Launcher wouldn't apply and Gw2Launcher would need to be started as an administrator.

Not through Steam? See below.

> Will there be a way to start a non-steam account via Gw2Launcher and yet display the game on steam profile status?

For GW2 to connect with Steam, either GW2 has to load Steam (-provider Steam) or Steam has to load GW2. To force it would require modding GW2 or faking it - it'd _only_ allow Steam to track when you're playing. The laziest option would be launching an account through Steam and just letting it sit on the launcher to track the play time.

**Comment by writzx** [comment]
> To force it would require modding GW2 or faking it - it'd only allow Steam to track when you're playing. 

I was hinting more towards the third option of launching the steam game through Gw2Launcher using the steam protocol. Something like:
```
steam steam://rungameid/1284210//-shareArchive
```

(although I believe there is an extra prompt if you include the command line options here rather than in the steam launch options)

Although I am not sure how feasible it would be for Gw2Launcher to hook into the launched processes using that method.

**Comment by ovoon0** [comment]
Is it possible for steam account & non steam account to be launched at the same time?

**Comment by Healix** [comment]
> Is it possible for steam account & non steam account to be launched at the same time?

If you want to launch an account through Steam and launch another account outside of Steam, you have to use the -shareArchive option to allow multiple clients to share the same files (shortcut: steam.exe -applaunch 1284210 -sharearchive).

**Comment by MiyukoSenyo** [comment]
Hello ! First : thanks for your software !
I have a question, i use steam and non steam so i have 2 .exe its ok but when i launch the second acc, the first one lose like 100fps and even if i close the 2nd account i'm stuck at ~50 FPS instead of 165+.

Do you know why ? 

The goal is to have the first acc on my UW screen while having the second on windowed for play it when waiting meta etc. so maybe full ultra 3440x1440 is the prob ?

Sorry for bad english and thanks for your answer :)

**Comment by Healix** [comment]
Ensure GW2 is using your GPU and it's operating at full power. Using an integrated GPU or your GPU entering a power saving mode will severely impact your FPS.

Note you don't need 2 installs. You can launch Steam from the original install or the original from Steam's - they're the same files.

**Comment by MiyukoSenyo** [comment]
Hm just checked everything is fine power side and on the paper, i should dont have any pb launching 2 games (GPU 7900xtx, CPU 7800x3D, 32Go ram, etc.) its really weird.

I was thinking using 2 installation would be more "optimized" because not using 2 times the same thing. I'll delete one then ^^

I tried tonight and still the same problem. The second one windowed with medium setting run perfectly fine and the first one full ultra setting 3440x1440 is dying inside. I think i'll not be able to do the initial plan. 
I'll check tomorrow with lowest setting/resolution

Thank for your answer :)
